### PR TITLE
G2 - Tic Tac Toe

### PR DESCRIPTION
**Pull request by seg2151**
Members:
Sebatian
Rahul

### CODE REVIEW

File: tic_tac_toe/game.py [hunk 1]
```diff
@@ -24,7 +28,14 @@ def _position_is_valid(position):
 
     Returns True if given position is valid, False otherwise.
     """
-    pass
+    valid_positions = [
```
Comment (koakekuna) [inline tic_tac_toe/game.py:31]: nice visual representation of what positions are valid! much better than checking if len(position) == 2, etc...

File: tic_tac_toe/game.py [hunk 2]
```diff
@@ -47,7 +61,13 @@ def _is_winning_combination(board, combination, player):
     Returns True of all three positions in the combination belongs to given
     player, False otherwise.
     """
-    pass
+    #(0,0)(0,1)(0,2)
+    #board[0][0], board [0][1], board[0][2]
+    #board[combination[0][0]][combination[0][1]], board[combination[1][0]][combination[1][1]], board[combination[2][0]][combination[2][1]]
+    
+    if (player,player,player) == (board[combination[0][0]][combination[0][1]], board[combination[1][0]][combination[1][1]], board[combination[2][0]][combination[2][1]]):
+        return True
```
Comment (koakekuna) [inline tic_tac_toe/game.py:69]: incredible you did it in one line

File: tic_tac_toe/game.py [hunk 3]
```diff
@@ -87,18 +126,68 @@ def move(game, player, position):
...
+                
+                
+    board_template = """
+{0}  |  {1}  |  {2}
+--------------
+{3}  |  {4}  |  {5}
+--------------
+{6}  |  {7}  |  {8}
+""" 
+
```
Comment (koakekuna) [inline tic_tac_toe/game.py:184]: I like this board_template much better too. Again, much more visual, so easier to maintain.

File: tic_tac_toe/game.py (hunk 2 above)
Comment (seg2151) [reply tic_tac_toe/game.py:69]: thanks! I was a bit worried about the readability but it felt very good to keep it all in one line =P

**Comment by urshala** [comment]
This is so beautifully done. All the variable names are meaningful. The space around the lines make it easy to follow the code. 

Thank you for this , I have so much to learn :)

**Pull request closed without merging**